plugin/utils/shell.py

```diff
@@ -86,6 +87,183 @@ def create_lods():
 	return msgs
 
 
+def generate_rig_edit():
+    """Automatic rig edit generator by NDP. Detects posed bones and automatically generates nodes and offsets them."""
+    msgs = []
+    #logging.info(f"Generating rig edit")
+    print("-------------------------------------------------------------")
+    print("Initiating rig edit")
+    
+    #Check if selected object is a valid armature
+    if bpy.context.active_object.type != 'ARMATURE':
+        #Object is not an armature. Cancelling.
+        print("No armature selected.")
+        msgs.append(f"No armature selected.")
+        return msgs
+    
+    #Store current mode
+    Mode = bpy.context.mode
+    
+    #Store number of edits
+    Edits = 0
+
+    #Force Switch to pose mode.
+    bpy.ops.object.mode_set(mode='POSE')
+    
+    #Get the armature
+    armature = bpy.context.object
+    
+    #Iterate over every bone in the armature.
+    for i in armature.pose.bones:
```

Comment:
Use better var name. A typical convention is `p_bone` for pose bones

Reply:
changed to p_bone